src/esp/physics/PhysicsManager.cpp

```diff
@@ -79,9 +79,15 @@ int PhysicsManager::addObject(const std::string& configFileHandle,
   if (attachmentNode == nullptr) {
     objectNode = &staticSceneObject_->node().createChild();
   }
-
-  bool objectSuccess =
-      makeAndAddRigidObject(nextObjectID_, physicsObjectAttributes, objectNode);
+  // verify whether necessary assets exist, and if not, instantiate them
+  bool instanceSuccess =
+      resourceManager_.instantiateAssetsOnDemand(configFileHandle);
+
+  bool objectSuccess = false;
+  if (instanceSuccess) {  // only make object if asset instantiation succeeds
+    objectSuccess = makeAndAddRigidObject(nextObjectID_,
+                                          physicsObjectAttributes, objectNode);
+  }
```

Comment:
```suggestion
  // verify whether necessary assets exist, and if not, instantiate them;
  // only make object if asset instantiation succeeds
  bool objectSuccess = resourceManager_.instantiateAssetsOnDemand(configFileHandle) &&
    makeAndAddRigidObject(nextObjectID_,
                          physicsObjectAttributes, objectNode);
  }
```

Make use of the short-circuit evaluation here? Tho could be cryptic for people not used to such workflow.